mava/systems/tf/maddpg/execution.py

```diff
@@ -13,30 +13,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""MADDPG system executor implementation."""
-
-from typing import Dict, List, Optional, Tuple, Union
+import copy
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import dm_env
+import numpy as np
 import sonnet as snt
 import tensorflow as tf
 import tensorflow_probability as tfp
 from acme import types
 from acme.specs import EnvironmentSpec
+
+# Internal imports.
 from acme.tf import utils as tf2_utils
 from acme.tf import variable_utils as tf2_variable_utils
 from dm_env import specs
+from numpy.random import randint
 
 from mava import adders
 from mava.systems.tf import executors
+from mava.utils.sort_utils import sort_str_num
 
 Array = specs.Array
 BoundedArray = specs.BoundedArray
 DiscreteArray = specs.DiscreteArray
-
 tfd = tfp.distributions
 
 
+def sample_new_agent_keys(
```

Comment:
Is this only applicable to MADDPG? If not, perhaps we can move it to a utils file or something similar?

Reply:
Good point. In the future, this will hopefully be applicable to all the executors. I moved it to the sort_utils.